mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/JobClusterActor.java

```diff
@@ -2180,35 +2185,68 @@ public void onJobClusterUpdateArtifact(UpdateJobClusterArtifactRequest artifactR
                     .withUploadedAt(System.currentTimeMillis())
                     .build();
 
-            JobClusterDefinitionImpl updatedDefn = new JobClusterDefinitionImpl.Builder().from(jobClusterMetadata.getJobClusterDefinition())
-                    .withJobClusterConfig(newConfig)
-                    .build();
-            IJobClusterMetadata jobCluster = new JobClusterMetadataImpl.Builder()
-                    .withIsDisabled(jobClusterMetadata.isDisabled())
-                    .withLastJobCount(jobClusterMetadata.getLastJobCount())
-                    .withJobClusterDefinition(updatedDefn)
-                    .build();
-
-            updateAndSaveJobCluster(jobCluster);
-
-            sender.tell(new UpdateJobClusterArtifactResponse(artifactReq.requestId, SUCCESS, name + " artifact updated"), getSelf());
-
-            eventPublisher.publishAuditEvent(
-                    new LifecycleEventsProto.AuditEvent(LifecycleEventsProto.AuditEvent.AuditEventType.JOB_CLUSTER_UPDATE,
-                        jobClusterMetadata.getJobClusterDefinition().getName(),
-                        name + " artifact update")
-            );
+            updateJobClusterConfig(newConfig);
             if(!artifactReq.isSkipSubmit()) {
                 getSelf().tell(new SubmitJobRequest(name,artifactReq.getUser(), (empty())), getSelf());
             }
-
+            sender.tell(new UpdateJobClusterArtifactResponse(artifactReq.requestId, SUCCESS, name + " artifact updated"), getSelf());
         } catch(Exception e) {
             logger.error("job cluster not updated ", e);
             sender.tell(new UpdateJobClusterArtifactResponse(artifactReq.requestId, SERVER_ERROR, name + " Job cluster artifact updation failed " + e.getMessage()), getSelf());
         }
         if(logger.isTraceEnabled()) { logger.trace("Exit JobClusterActor:onJobClusterUpdateArtifact"); }
     }
 
+    private void updateJobClusterConfig(JobClusterConfig newConfig) throws Exception {
+        JobClusterDefinitionImpl updatedDefn = new JobClusterDefinitionImpl.Builder().from(jobClusterMetadata.getJobClusterDefinition())
+            .withJobClusterConfig(newConfig)
+            .build();
+        IJobClusterMetadata jobCluster = new JobClusterMetadataImpl.Builder()
+            .withIsDisabled(jobClusterMetadata.isDisabled())
+            .withLastJobCount(jobClusterMetadata.getLastJobCount())
+            .withJobClusterDefinition(updatedDefn)
+            .build();
+
+        updateAndSaveJobCluster(jobCluster);
+
+        eventPublisher.publishAuditEvent(
+            new LifecycleEventsProto.AuditEvent(LifecycleEventsProto.AuditEvent.AuditEventType.JOB_CLUSTER_UPDATE,
+                jobClusterMetadata.getJobClusterDefinition().getName(),
+                name + " artifact update")
+        );
+    }
+
+    @Override
+    public void onJobClusterUpdateSchedulingInfo(UpdateSchedulingInfo request) {
```

Comment:
nit: validate UpdateSchedulingInfo to ensure versoin/schedulingInfo parts are present.